Classes/IndexQueue/RecordMonitor.php

```diff
@@ -45,13 +43,6 @@ class RecordMonitor extends AbstractDataHandlerListener
 {
 
     /**
-     * Solr TypoScript configuration
-     *
-     * @var TypoScriptConfiguration
-     */
-    protected $solrConfiguration;
```

Comment:
I would agree to Thomas because by now we assume that solrConfiguration is initialized (as a side effect). We can only get the configuration, when are able to detect a context page (to retrieve the configuration). By passing it (internally for protected function) we force users to make create a configuration. I think we can refactor this later and get rid of it at some parts e.g. in getRecord, but by now i would say it's better to force it instead to relly on an side effect.